patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -1555,11 +1555,15 @@
  			for (int m = 3; m < 10; m++) {
  				if (armor[m].wingSlot > 0 && IsAValidEquipmentSlotForIteration(m)) {
  					if (!hideVisibleAccessory[m] || (velocity.Y != 0f && !mount.Active))
-@@ -9305,12 +_,15 @@
+@@ -9305,12 +_,19 @@
  					wingsLogic = armor[m].wingSlot;
  				}
  			}
 +			*/
++
++			for (int n = 10; n < 13; n++) {
++				ApplyEquipVanity(n, armor[n]);
```

Comment:
RE: Verify no reason to call vanilla vanity accessory updates when looping through the proper armor slots.
If no need, than just call itemloader.updatevanity and reduce # of calls

Reply:
Fair, should make more sense now 👍

Reply:
Per discord, add a one line comment above the added for loop describing why it's there.
It would be good to drop a line confirming the "hideVisibility" array goes unused for slots 10-12 for sanity's sake, although I'd swear its unused.

Can merge when comment is on there, ping Mirsario in discord when done.

Some context:
Discord discussed if needs to check hideVisibility, but given this is helm, chest, greaves, don't believe need to check for the hook as player doesn't set it. Don't need to check accessory compat either, accessories control their compat, just came up as confusion